plugins/commands.js

```diff
@@ -0,0 +1,183 @@
+            notes: 'Get a script or binary of specific command',
+            tags: ['api', 'commands'],
+            plugins: {
+                'hapi-swagger': {
+                    security: [{ token: [] }]
+                }
+            },
+            handler: (request, reply) => {
+                const namespace = request.params.namespace;
+                const name = request.params.name;
+                const version = request.params.version;
+
+                const id = `${namespace}-${name}-${version}`;
+
+                cache.get(id, (err, value) => {
+                    if (err) {
+                        return reply(err);
+                    }
+                    if (!value) {
+                        return reply(boom.notFound());
+                    }
+
+                    // @TODO put cache headers in here
+                    const response = reply(Buffer.from(value.c.data));
+
+                    response.headers = value.h;
+
+                    return response;
+                });
+            },
+            validate: {
+                params: {
+                    namespace: SCHEMA_COMMAND_NAMESPACE,
+                    name: SCHEMA_COMMAND_NAME,
+                    version: SCHEMA_COMMAND_VERSION
+                }
+            }
+        }
+    }, {
+        method: 'PUT',
+        path: '/commands/{namespace}/{name}/{version}',
+        config: {
+            description: 'Write command',
+            notes: 'Write a script or binary of specific command',
+            tags: ['api', 'commands'],
+            payload: {
+                maxBytes: parseInt(options.maxByteSize, 10) || DEFAULT_BYTES,
+                parse: false
+            },
+            auth: {
+                strategies: ['token'],
+                scope: ['build']
+            },
+            plugins: {
+                'hapi-swagger': {
+                    security: [{ token: [] }]
+                }
+            },
+            handler: (request, reply) => {
+                const pipelineId = request.auth.credentials.pipelineId;
+                const namespace = request.params.namespace;
+                const name = request.params.name;
+                const version = request.params.version;
+                const id = `${namespace}-${name}-${version}`;
+                const apihost = request.server.app.ecosystem.api;
+                const contents = {
+                    c: request.payload,
+                    h: {}
+                };
+                const size = Buffer.byteLength(request.payload);
+
+                // Store all x-* and content-type headers
+                Object.keys(request.headers).forEach((header) => {
+                    if (header.indexOf('x-') === 0 || header === 'content-type') {
+                        contents.h[header] = request.headers[header];
+                    }
+                });
+
+                return canPublishCommand(namespace, name, pipelineId, apihost)
```

Comment:
Actually you don't need this function. The build jwt token contains the pipelineId inside.
https://github.com/screwdriver-cd/models/blob/d71bacb381536e4298049f398e3c787901799db0/lib/build.js#L162